tests/linalg_test.py

```diff
@@ -783,6 +783,37 @@ def testMultiDot(self, shapes, dtype, rng_factory):
     self._CompileAndCheck(jnp_fun, args_maker, check_dtypes=True,
                           atol=tol, rtol=tol)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+       "_lhs={}_rhs={}_rcond={}".format(
+           jtu.format_shape_dtype_string(lhs_shape, dtype),
+           jtu.format_shape_dtype_string(rhs_shape, dtype),
+           rcond),
+       "lhs_shape": lhs_shape, "rhs_shape": rhs_shape, "dtype": dtype,
+       "rcond": rcond, "rng_factory": rng_factory}
+      for lhs_shape, rhs_shape in [
+          ((1, 1), (1, 1)),
+          ((4, 6), (4,)),
+          ((6, 6), (6, 1)),
+          ((8, 6), (8, 4)),
```

Comment:
I think it would also be a good idea to test co-linear and/or duplicate columns in the design matrix `a`, which you won't get with random numbers. I think these would correspond to repeated singular values in the SVD? (repeated singular values will probably break the SVD JVP rule)

Reply:
Done